sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1271,6 +1271,7 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
         For each categorical feature, there must be at most `max_bins` unique
         categories, and each categorical value must be less then `max_bins - 1`.
         Negative values for categorical features are treated as missing values.
+        Internally integers are cast as float array and hence 1.0 and 1 is treated as the same. 
```

Comment:
Given that this is user facing, I think it's better to say what the behavior is without going into the internals:

```suggestion
        Categorical values inputted as integers are treated the same as their floating
        point counterpart. For example, 1.0 and 1 are treated as the same category.
```

Reply:
How about "All categorical values are converted to floating point numbers. This means that categorical values of 1.0 and 1 are treated as the same category." - my attempt at avoiding the word "inputted", which for some reason I am not mad keen on.

Reply:
@betatim I am okay with your suggestion.